### ISSUE TITLE
Mode change string failing to parse on clients

### ISSUE THREAD

**Issue by adamus1red** [inspircd/inspircd#1636]
**Description**

Mode change string failing to parse on clients.

The recent 3.0 update breaks the nicklist update on many IRC clients (Confirmed HexChat & have had it reported as affecting mIRC)

Reported upstream to hexchat (https://github.com/hexchat/hexchat/issues/2335) and issue appears to be caused by the MODE change having the `:` without any spacing before the nickname
i.e. `MODE #test +o :Guest3` Vs `MODE #test +o : Guest3` 

**Steps to reproduce the issue:**

1. User joins channel
2. Recieve mode change (i.e. +o, +h, etc)

**Describe the results you received:**
Client side nicklist on client is not updated with status change.

**Describe the results you expected:**
Client side nick list is updated with correct status change

**Additional information you deem important (e.g. issue happens only occasionally):**
Hexchat message log
```
>> @time=2019-05-06T15:07:29.000Z :Guest3!~a@testnet-qs3.dod.0sg3ho.IP PART :#test
>> @time=2019-05-06T15:07:29.000Z :Guest3!~a@testnet-qs3.dod.0sg3ho.IP JOIN #test * :realname
<< MODE #test
<< WHO #test %chtsunfra,152
>> @time=2019-05-06T15:07:29.000Z :irc.deb1.lan 332 Guest3 #test :Feel free to test any bots, scripts, etc in this channel.
>> @time=2019-05-06T15:07:29.000Z :irc.deb1.lan 333 Guest3 #test admin :1403572744
>> @time=2019-05-06T15:07:29.000Z :irc.deb1.lan 353 Guest3 = #test :Guest3 @God @^Quark^
>> @time=2019-05-06T15:07:29.000Z :irc.deb1.lan 366 Guest3 #test :End of /NAMES list.
>> @time=2019-05-06T15:07:29.000Z :God!God@the.all.mighty.one NOTICE Guest3 :[#test] Welcome to #test. Feel free to test any bots, scripts, etc in this channel.
>> @time=2019-05-06T15:07:29.000Z :God!God@the.all.mighty.one MODE #test +o :Guest3
>> @time=2019-05-06T15:07:29.000Z :irc.deb1.lan 324 Guest3 #test :+Pnt
>> @time=2019-05-06T15:07:29.000Z :irc.deb1.lan 329 Guest3 #test :1403572555
>> @time=2019-05-06T15:07:29.000Z :irc.deb1.lan 354 Guest3 152 #test ~a testnet-qs3.dod.0sg3ho.IP irc.deb1.lan Guest3 H@s 0 :realname
>> @time=2019-05-06T15:07:29.000Z :irc.deb1.lan 354 Guest3 152 #test God the.all.mighty.one irc.deb1.lan God H@ 0 :Also known as Satan
>> @time=2019-05-06T15:07:29.000Z :irc.deb1.lan 354 Guest3 152 #test ~quantum quark.lan irc.deb1.lan ^Quark^ H@s ^Quark^ :Quarkly
>> @time=2019-05-06T15:07:29.000Z :irc.deb1.lan 315 Guest3 #test :End of /WHO list.
<< PING LAG2327093070
>> @time=2019-05-06T15:07:32.000Z :irc.deb1.lan PONG irc.deb1.lan :LAG2327093070
<< NAMES #test
>> @time=2019-05-06T15:07:57.000Z :irc.deb1.lan 353 Guest3 = #test :@Guest3 @God @^Quark^
>> @time=2019-05-06T15:07:57.000Z :irc.deb1.lan 366 Guest3 #test :End of /NAMES list.
<< WHO #test %chtsunfra,152
<< PING LAG2327123071
>> @time=2019-05-06T15:08:02.000Z :irc.deb1.lan 354 Guest3 152 #test ~a testnet-qs3.dod.0sg3ho.IP irc.deb1.lan Guest3 H@s 0 :realname
>> @time=2019-05-06T15:08:02.000Z :irc.deb1.lan 354 Guest3 152 #test God the.all.mighty.one irc.deb1.lan God H@ 0 :Also known as Satan
>> @time=2019-05-06T15:08:02.000Z :irc.deb1.lan 354 Guest3 152 #test ~quantum quark.lan irc.deb1.lan ^Quark^ H@s ^Quark^ :Quarkly
>> @time=2019-05-06T15:08:02.000Z :irc.deb1.lan 315 Guest3 #test :End of /WHO list.
>> @time=2019-05-06T15:08:02.000Z :irc.deb1.lan PONG irc.deb1.lan :LAG2327123071
<< PING LAG2327153071
```

Workaround of running a manual `/names` fixes the issue, until someone else joins.

**Output of `./bin/inspircd --version`:**
InspIRCd-3.0.0

**Comment by SaberUK** [comment]
This is a HexChat issue (which is why they fixed it) as the messages we send are entirely valid according to the message rules in RFC 1459.

This is also unfortunately not something we can really work around because of the changes in how we emit messages which were necessary for tag support.

**Comment by sacredland** [comment]
I got the same trouble, only with inspircd v3. Why you need client to get fixed, when they work well on other ircd? It's clearly due to something wrong into inpsircd v3. 

On androirc same thing happen, but when you do /names or /list it's duplicate the nickname list and adding status to user, so you have @nick and nick into the nick list.. just duplication. with and without status. 

And "you better code your clients to fix our mistake" or "it's a client issue" isn't a good answer, as it is not an issue on any other ircd (ircu [even from undernet!!], nafarious2, bahamut, ratbox, plexus, charybdis). I hope you don't think this is a viable way to promote inspircd v3.

**Comment by SaberUK** [comment]
>. Why you need client to get fixed, when they work well on other ircd?

Just because it happens to work with other IRC server implementations doesn't mean its not broken.

The parsing rules in RFC 1459 are very clear about how messages should be parsed and InspIRCd emits messages which are valid according to those rules.

**Comment by sacredland** [comment]
IRC is dying, and that kind of answer & management add another nail into the coffin. Thanks to PIA/LTM

"Maybe use one of those instead?"  No, I'll just switch IRCds instead. It's simpler, and user wont be affected by the kind of knowed bug.

**Comment by SaberUK** [comment]
Update: this is now fixed in HexChat 2.14.3.
